### PR TITLE
UX: Hide "Create Tag" option if user cannot create tag.

### PR DESCRIPTION
**Pull request by udan11**

### CODE REVIEW

File: app/serializers/basic_category_serializer.rb [hunk 1]
```diff
@@ -65,4 +68,28 @@ def custom_fields
   def include_custom_fields?
     custom_fields.present?
   end
+
+  def include_allowed_tags?
+    SiteSetting.tagging_enabled
+  end
+
+  def allowed_tags
+    object.tags.pluck(:name)
```
Comment (SamSaffron) [inline app/serializers/basic_category_serializer.rb]: Is this adding an N+1 to the topic list? very concerned about a change here.

**Comment by ZogStriP** [comment]
Any news about this @udan11?

**Comment by SamSaffron** [comment]
I approve of the change but am uneasy with the implementation. 

So we really need to carry 3 flavors of category serialization? I worry here that we can introduce code drift this way even with the mixin.

**Comment by udan11** [comment]
@SamSaffron, I could use the `CategorySerializer`, which would prevent future code drift at the cost of having bigger payloads.

**Pull request merged**